arrow-cpp: 0.15.1 -> 0.16.0

tobim opened the pull request:
Changelog: https://arrow.apache.org/release/0.16.0.html

This switches to explicitly enabling dependency lookups for optional features (they were not picked up before).
It also enables building of the Parquet command line tools.

<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

tobim committed:
arrow-cpp: 0.15.1 -> 0.16.0

This switches to explicitly enabling dependency lookups for
optional features (they were not picked up before).
It also enables building of the Parquet command line tools.

veprbl commented:
@GrahamcOfBorg build arrow-cpp
@GrahamcOfBorg build pkgsStaic.arrow-cpp
@GrahamcOfBorg build python37Packages.arrow-cpp
@GrahamcOfBorg build python38Packages.arrow-cpp

veprbl commented:
@GrahamcOfBorg build pkgsStatic.arrow-cpp

veprbl commented:
@GrahamcOfBorg build python37Packages.pyarrow
@GrahamcOfBorg build python38Packages.pyarrow

veprbl commented:
Looks like this breaks pyarrow and the jemalloc-disable-shared.patch needs to be updated too.

tobim committed:
pytest-lazy-fixture: init at 0.6.3

veprbl commented:
@GrahamcOfBorg build pkgsStatic.arrow-cpp
@GrahamcOfBorg build python37Packages.pyarrow
@GrahamcOfBorg build python38Packages.pyarrow

veprbl commented on pkgs/development/libraries/arrow-cpp/default.nix:
```diff
@@ -35,6 +35,7 @@ in stdenv.mkDerivation rec {
   patches = [
     # patch to fix python-test
     ./darwin.patch
+    ./use-upstream-target-names.patch
```
This has to do with the enabling of CMAKE_FIND_PACKAGE_PREFER_CONFIG? We need CMAKE_FIND_PACKAGE_PREFER_CONFIG for the static build? Can we upstream this patch?

tobim replied:
I was planning to, just didn't get around to it yet.

tobim replied:
Upstreaming at https://github.com/apache/arrow/pull/6480.

jonringer reviewed (approved):
diff LGTM

failures are broken on master
```
[15 built (4 failed), 0.0 MiB DL]
error: build of '/nix/store/fqr9xrpbj4qysrkml9v8b72haa4fc759-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/79817
4 package failed to build:
python37Packages.google_cloud_bigquery python37Packages.ibis-framework python38Packages.google_cloud_bigquery python38Packages.ibis-framework

16 package built:
arrow-cpp python27Packages.awkward python27Packages.pyarrow python27Packages.pytest-lazy-fixture python27Packages.uproot python27Packages.uproot-methods python37Packages.awkward python37Packages.pyarrow python37Packages.pytest-lazy-fixture python37Packages.uproot python37Packages.uproot-methods python38Packages.awkward python38Packages.pyarrow python38Packages.pytest-lazy-fixture python38Packages.uproot python38Packages.uproot-methods
```

jonringer commented:
@GrahamcOfBorg build arrow-cpp python27Packages.awkward python27Packages.pyarrow python27Packages.pytest-lazy-fixture python27Packages.uproot python27Packages.uproot-methods python37Packages.awkward python37Packages.pyarrow python37Packages.pytest-lazy-fixture python37Packages.uproot python37Packages.uproot-methods python38Packages.awkward python38Packages.pyarrow python38Packages.pytest-lazy-fixture python38Packages.uproot python38Packages.uproot-methods

veprbl commented on pkgs/development/libraries/arrow-cpp/default.nix:
```diff
@@ -35,6 +35,7 @@ in stdenv.mkDerivation rec {
   patches = [
     # patch to fix python-test
     ./darwin.patch
+    ./use-upstream-target-names.patch
```
```suggestion
    (fetchpatch {
      url = "https://github.com/apache/arrow/commit/396861b38d2f4e805db7c2ecd2c96fff0ca2678b.patch";
      sha256 = "19hdm9ampq8g2swnyfgq7im4pqa55yy4ni86fjrb9yv5jzwvx0b1";
    })
```

veprbl commented on pkgs/development/libraries/arrow-cpp/default.nix:
```diff
@@ -40,6 +40,15 @@ in stdenv.mkDerivation rec {
+  # Adjust CMake target names to make -DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON work.
+  # Remove this when updating to the next version.
+  postPatch = ''
+    patch -p2 < ${(fetchpatch {
```
Just use
```
    (fetchpatch {
      url = "https://github.com/apache/arrow/commit/396861b38d2f4e805db7c2ecd2c96fff0ca2678b.patch";
      sha256 = "0ki7nx858374anvwyi4szz5hgnnzv4fghdd05c38bzry9rfljgb1";
      stripLen = 1;
    })
```

tobim replied:
Nice, I didn't know about that.

veprbl committed:
pythonPackages.pytest-lazy-fixture: use pytestCheckHook

veprbl committed:
arrow-cpp: fix RPATH on darwin

Instead of patching test binaries with install_name_tool just set
CMAKE_SKIP_BUILD_RPATH to OFF let cmake compile them with RPATH set
correctly.

Add CMAKE_INSTALL_RPATH to set correct RPATH for the installed
executables for parquet tools. We could set it to an absolute path using
"-DCMAKE_INSTALL_RPATH=${placeholder "out"}", but instead we use
@loader_path to provide a relocable solution.

veprbl commented:
@GrahamcOfBorg build pkgsStatic.arrow-cpp
@GrahamcOfBorg build python37Packages.pyarrow
@GrahamcOfBorg build python38Packages.pyarrow

veprbl commented:
@tobim Thank you!
